pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -849,6 +849,63 @@ func (m *kubeGenericRuntimeManager) purgeInitContainers(ctx context.Context, pod
 	}
 }
 
+// findNextInitContainerToRun returns the status of the last failed container, the
+// index of next init container to start, or done if there are no further init containers.
+// Status is only returned if an init container is failed, in which case next will
+// point to the current container.
+func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus) (status *kubecontainer.Status, next *v1.Container, done bool) {
```

Comment:
alright, so this is the method that has been recovered from 1.27 to fix the regression